File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for i in s:
            if i=='(' or i=='{' or i=='[':
                stack.append(i)
            else:
                if not stack:
                    return False
                elif i==')' and stack[-1]=='(':
                    stack=stack[:-1]
                elif i==']' and stack[-1]=='[':
                    stack=stack[:-1]
                elif i=='}' and stack[-1]=='{':
                    stack=stack[:-1]
                else:
                    return False
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.c
bool isValid(char* s) {
    char st[10000];
    int top=-1;
    for(int i=0;s[i]!='\0';i++){
        if(s[i]=='('||s[i]=='{'||s[i]=='[')st[++top]=s[i];
        else{
            if(top==-1)return false;
            if(st[top]=='('&&s[i]==')')top--;
            else if(st[top]=='{'&&s[i]=='}')top--;
            else if(st[top]=='['&&s[i]==']')top--;
            else return false;
        }
    }
    if(top==-1)return true;
    return false;
}